Documentation/installation/k8s-install-migration.rst

```diff
@@ -0,0 +1,340 @@
+
+It is strongly recommended to install Cilium using the :ref:`cluster-pool <ipam_crd_cluster_pool>`
+IPAM allocator. This provides the strongest assurance that there will
+be no IP collisions.
+
+.. warning::
+  Migration is highly dependent on the exact configuration of existing
+  clusters. It is, thus, strongly recommended to perform a trial migration
+  on a test or lab cluster.
+
+Overview
+========
+
+The migration process utilizes the :ref:`per-node configuration<per-node-configuration>`
+feature to selectively enable Cilium CNI. This allows for a controlled rollout
+of Cilium without disrupting existing workloads.
+
+Cilium will be installed, first, in a mode where it establishes an overlay
+but does not provide CNI networking for any pods. Then, individual nodes will
+be migrated.
+
+In summary, the process looks like:
+
+1. Install cilium in "secondary" mode
+2. Cordon, drain, migrate, and reboot each node
+3. Remove the existing network provider
+4. (Optional) Reboot each node again
+
+
+Migration procedure
+===================
+
+Preparation
+-----------
+
+- Optional: Create a :ref:`Kind <gs_kind>` cluster and install `Flannel <https://github.com/flannel-io/flannel>`_ on it.
+
+    .. parsed-literal::
+
+      $ cat <<EOF > kind-config.yaml
+      apiVersion: kind.x-k8s.io/v1alpha4
+      kind: Cluster
+      nodes:
+      - role: control-plane
+      - role: worker
+      - role: worker
+      networking:
+        disableDefaultCNI: true
+      EOF
+      $ kind create cluster --config=kind-config.yaml
+      $ kubectl apply -n kube-system --server-side -f \ |SCM_WEB|\/examples/misc/migration/install-reference-cni-plugins.yaml
+      $ kubectl apply --server-side -f https://github.com/flannel-io/flannel/releases/latest/download/kube-flannel.yml
+      $ kubectl wait --for=condition=Ready nodes --all
+
+- Optional: Monitor connectivity.
+
+  You may wish to install a tool such as `goldpinger <https://github.com/bloomberg/goldpinger>`_
+  to detect any possible connectivity issues.
+
+1.  Select a **new** CIDR for pods. It must be distinct from all other CIDRs in use.
+
+    For Kind clusters, the default is ``10.244.0.0/16``. So, for this example, we will
+    use ``10.245.0.0/16``.
+
+2.  Select a **distinct** encapsulation port. For example, if the existing cluster
+    is using VXLAN, then you should either use GENEVE or configure Cilium to use VXLAN
+    with a different port.
+
+    For this example, we will use VXLAN with a non-default port of 8473. But the Cilium
+    default of GENEVE encapsulation is preferable.
+
+3.  Create a helm ``values-migration.yaml`` file based on the following example. Be sure to fill
+    in the CIDR you selected in step 1.
+
+    .. code-block:: yaml
+
+        operator:
+          unmanagedPodWatcher:
+            restart: false # Migration: Don't restart unmigrated pods
+        tunnel: vxlan # Migration: Optional: default is GENEVE, configure as needed
```

Comment:
Is Geneve definitely the default? Helm default value is VXLAN:
https://github.com/cilium/cilium/blob/c9723a8df3cfa336da1f8457a864105d8349acfe/install/kubernetes/cilium/values.yaml#L1681

Reply:
Oh, you're totally right. Whoops.